File: Estudos solo/Python/exercicios atividade 7 em python/at7ex2.py

def rev_str(str, lenght):
    letra = str[lenght - 1]
    if (lenght == 0):
        str_rev = ""
        return str_rev
    else:
        str_rev = letra + rev_str(str, lenght - 1)
        return str_rev


string = input("String: ")

lenght = len(string)

print(rev_str(string, lenght))

File: ATP 2/Codigos/C e C++/AULA 8/at7_ex2.cpp
#include <stdio.h>
#include <stdlib.h>
#include <ctype.h>
#include <string.h>

int str_rev(char *str, int lenght)
{
    printf("%c", str[lenght]);
    if(lenght == 0)
    {
        return 0;
    }
    return(str_rev(str, lenght-1));
}

int main()
{
    char str[11]= "merda mole";
    str_rev(str, strlen(str));

    getchar();
    return 0;
}
